ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -109,6 +109,9 @@ public KsqlEngine(final KsqlConfig ksqlConfig, final KafkaTopicClient topicClien
     this.persistentQueries = new HashMap<>();
     this.liveQueries = new HashSet<>();
     this.functionRegistry = new FunctionRegistry();
+    SchemaRegistryClientFactory.setSchemaRegistryClient((String) ksqlConfig.get(KsqlConfig
```

Comment:
I don't quite understand why we need a `SchemaRegistryClientFactory`? Why do we not just have a single `SchemaRegistryClient`?

Reply:
For the engine we will have one schema registry but we also have the option of choosing another schema registry by user with passing it through the session properties.